# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/test/test_bug1101364.html
# Commit: 9b15baab6f77
# Full Hash: 9b15baab6f77d09eca2956276f66fee97555e4a1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 3: Make `HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument` and `HTMLEditor::InitEditorContentAndSelection` do nothing if the document is partially editable r=m_kato
#   
#   They and their callees work with the result of `GetRoot()` which is the document
#   element or the body element.  If the body is not editable, `Selection` should
#   not be updated in non-editable region nor `<br>` elements should not be
# ==============================================================================

diff -r 4d1e93c629da -r 9b15baab6f77 dom/base/test/test_bug1101364.html
--- a/dom/base/test/test_bug1101364.html	Thu Sep 22 06:24:38 2022 +0000
+++ b/dom/base/test/test_bug1101364.html	Thu Sep 22 06:27:37 2022 +0000
@@ -26,42 +26,67 @@
 <pre id="test">
 <script class="testbody" type="text/javascript">
 
-async function test()
-{
-  var iframe1 = document.getElementById('test1');
-  iframe1.focus();
-  var docShell = SpecialPowers.wrap(iframe1.contentWindow).docShell;
+SimpleTest.waitForExplicitFinish();
+SimpleTest.waitForFocus(async () => {
+  await (async () => {
+    const iframe = document.getElementById("test1");
+    iframe.focus();
+    const docShell = SpecialPowers.wrap(iframe.contentWindow).docShell;
 
-  // test1
-  docShell.doCommand("cmd_selectAll");
-  var withoutContenteditable = await snapshotWindow(iframe1.contentWindow);
-
-  iframe1.contentDocument.getElementById('testDiv').setAttribute('contentEditable', true);
-  docShell.doCommand("cmd_selectAll");
-  var withContenteditable = await snapshotWindow(iframe1.contentWindow);
-  dump(withoutContenteditable.toDataURL());
-  dump(withContenteditable.toDataURL());
+    docShell.doCommand("cmd_selectAll");
+    info(
+      "Waiting for getting screenshot of \"Select All\" without contenteditable..."
+    );
+    const withoutContenteditable = await snapshotWindow(iframe.contentWindow);
 
-  ok(compareSnapshots(withoutContenteditable, withContenteditable, true)[0], 'Select all should look identical');
+    iframe.contentDocument
+      .getElementById("testDiv")
+      .setAttribute("contentEditable", true);
+    docShell.doCommand("cmd_selectAll");
+    info(
+      "Waiting for getting screenshot of \"Select All\" in contenteditable..."
+    );
+    const withContenteditable = await snapshotWindow(iframe.contentWindow);
+    const result =
+      compareSnapshots(withoutContenteditable, withContenteditable, true);
+    ok(
+      result[0],
+      `Select all should look identical\ngot: ${
+        result[2]
+      }\nexpected: ${result[1]}`
+    );
+  })();
 
-  // test2
-  var iframe2 = document.getElementById('test2');
-  iframe2.focus();
-  var docShell = SpecialPowers.wrap(iframe2.contentWindow).docShell;
-  var test2Inner = iframe2.contentDocument.getElementById('test2Inner');
-  test2Inner.style.MozUserSelect = 'text';
-  docShell.doCommand("cmd_selectAll");
-  var withoutUserSelect = await snapshotWindow(iframe2.contentWindow);
+  await (async () => {
+    const iframe = document.getElementById("test2");
+    iframe.focus();
+    iframe.contentDocument.querySelector("div[contenteditable]").focus();
+    const docShell = SpecialPowers.wrap(iframe.contentWindow).docShell;
+    const test2Inner = iframe.contentDocument.getElementById("test2Inner");
+    test2Inner.style.MozUserSelect = "text";
+    docShell.doCommand("cmd_selectAll");
+    info(
+      "Waiting for getting screenshot of \"Select All\" in contenteditable (use-select: text)..."
+    );
+    const withoutUserSelect = await snapshotWindow(iframe.contentWindow);
 
-  test2Inner.style.MozUserSelect = 'none';
-  docShell.doCommand("cmd_selectAll");
-  var withUserSelect = await snapshotWindow(iframe2.contentWindow);
-  ok(compareSnapshots(withoutUserSelect, withUserSelect, true)[0], 'Editable fields should ignore user select style');
+    test2Inner.style.MozUserSelect = "none";
+    docShell.doCommand("cmd_selectAll");
+    info(
+      "Waiting for getting screenshot of \"Select All\" in contenteditable (use-select: none)..."
+    );
+    const withUserSelect = await snapshotWindow(iframe.contentWindow);
+    const result = compareSnapshots(withoutUserSelect, withUserSelect, true);
+    ok(
+      result[0],
+      `Editable fields should ignore user select style\ngot: ${
+        result[2]
+      }\nexpected: ${result[1]}`
+    );
+  })();
 
   SimpleTest.finish();
-}
-window.onload = function() { setTimeout(test, 0); };
-SimpleTest.waitForExplicitFinish();
+});
 </script>
 </pre>
 </body>